### ISSUE TITLE
Crash when attempting to link device

### ISSUE THREAD

**Issue by fulldecent** [signalapp/Signal-iOS#5705]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Describe here the issue that you are experiencing.

### Steps to reproduce

I am experiencing this crash also:

1. Get a iPhone (13)
2. turn on lockdown mode
3. Install Signal, open it
4. say no on popups for: notifications, local network, contacts
5. Get a mac (M1 MBP, 2021)
6. turn on lockdown mode
7. Install Signal
8. Open Signal, get QR code
9. Scan QR code from camera app on iPhone
10. It opens Signal
11. Signal says it needs a camera to link a device (even though you already deep linked into Signal)
12. say no on popup for camera
13. go to settings app to turn on camera
14. scan qr code again, it opens signal app
15. signal says it wants to scan qr code
16. click to scan qr code from inside signal app
17. say no to popup for microphone

Expected outcome:

1. Camera access is not required for Signal to link a device (using a QR scanned by operating system and deep linking to Signal)
2. If you do want to allow camera access, linking a device from inside the app should link the device

Actual outcome:

1. Deep links do not work
2. Unnecessary permissions are requested many times inside the app
3. Crashes and linking devices is not possible
4. iOS app crashes after step 17

### Device info

**Device**:

1. iPhone 13
2. MacBook Pro 14-inch, 2021

**iOS version**:

1. 16.7
2. 14.0 (23A344)

**Signal version:**

The latest version on iOS app store as of 223-11-16.

### Link to debug log

Email sent with subject: iOS Crash Log: GitHub issue #

And filename Signal-2023-11-16-184844.ips

**Comment by katlol** [comment]
Reproduced. iPhone 15 Pro running 17.1.1 and MacBook Air M2 running 14.1.1

**Comment by bmfores** [comment]
Reproduced. iPhone 13 Pro on 17.1.1, and Macbook Air  M1 2020 running 14.0.

**Comment by prashand** [comment]
iPhone 6s Plus, Software Version = 15.8(19H370)
Signal iOS version = 6.50.1.0
Signal Mac Desktop app version = 6.39.1 production (Apple silicon)

Mobile app crashes immediately after scanning the QR code from the desktop app

**Comment by dave7280** [comment]
Same here. ip 15 pro. 
Signal iOS version = 6.50.1.0

**Comment by Miguel-Signal** [comment]
The bug fix is being worked on. In the meantime, there is a workaround.

Go into Signal Settings and enable a PIN. You should be able to link a device afterwards. You can disable the PIN afterwards if you'd like.

**Comment by r6t** [comment]
Signal iOS 6.50.1.0 on iPhone 15 Pro Max
signal-desktop on NixOS, both the latest Flathub and Nixpkgs versions

Mobile app crashes immediately upon scanning QR code to link. Confirming the PIN workaround was successful.

**Comment by yashbedi** [comment]
Hey is this issue fixed I would like to work on this, 
Thanks.

**Comment by TheExtravagantHobo** [comment]
Pin workaround doesn't work for me.
Tried to add Signal to an iPad mini 5th (MUXH2LL/A) running iPadOS 17.2. Signal fresh download.
Scanned the QR from iPhone 13 Pro (MLR63LL/A) running iOS 17.2.1, Signal 6.54.0.8.
Both on same wifi.

Scanning the QR with the phone (which always had the Pin on) immediately crashes the iPad app.

Removed Pin -- same result
Re-added Pin -- same result


?
Thanks! Happy New Year.

**Comment by Miguel-Signal** [comment]
Hi @TheExtravagantHobo, could you please email support@signal.org with debug logs from both the phone and iPad, as well as a crash report from the iPad? 

To share a debug log from your iPhone, tap your profile, Settings, Help, Submit Debug Log, Copy Link.

To share debug logs from an unlinked iPad, quit then re-open Signal and tap 8 times on an empty area of the screen. You should see a prompt to submit debug logs.

Here's information on how to share a crash report:

Go to iOS Settings > Privacy (& Security) > Analytics (& Improvements) > Analytics Data.
Locate and select the .ips crash log for Signal. The logs will likely be named in the format: Signal(DateTime)(DeviceName).ips, but please also ignore logs that have 'wakeups' in the file name.
Tap the Share icon in the top corner.

**Comment by TheExtravagantHobo** [comment]
There’s no data on the iPhone for Signal. It doesn’t crash. The iPad shows up in its linked devices at the same time the iPad crashes. The iPad is then not linked, even though it shows up in the Signal app as a linked device, and you can add more copies of it to linked devices.
Here’s a fresh one:

￼


Respectfully;
Alex Sonne
808-800-1150
***@***.***

**Comment by jc-frosty** [comment]
Should I create a new issue if my iPad signal app crashes after hitting ‘finish linking’ and neither of my devices have lockdown mode on.  I have had the iPad app linked before though I don’t know if inactivity(happens to the Mac app a lot) or an update unlinked them.

**Comment by Miguel-Signal** [comment]
@jc-frosty 30 days of inactivity will unlink a secondary device wether it's an iPad or Desktop. 

Could you please write into support@signal.org so we can troubleshoot this further?

**Comment by simonholgate** [comment]
+1 on @jc-frosty comment. iPad unlinked (possibly due to inactivity).

Tried re linking iPad with Android (Pixel 8). iPad shows up as linked on the phone but iPad app crashes immediately after linking and begins linking process again after restart.

**Comment by simonholgate** [comment]
As a quick update - deleting the ipad app and reinstalling allowed me to relink.

**Comment by 8bignic8** [comment]
I have the issue on an Iphon12 if I scan an qr code from my macbook, the Iphone freezes in the settings link new device tab. 

If I want to directly connect to it from this tab, i cant change my cameras and because one is Broken, I cant switch to the second one and therefore I cant connect the two accounts. 
Please help :))

**Comment by Miguel-Signal** [comment]
@8bignic8 Could you please write into [support@signal.org](mailto:support@signal.org) with a debug log link from your iPhone so we can troubleshoot this further?

To share a debug log from Signal iOS, tap your profile, Settings, Help, Submit Debug Log, Copy Link.

**Comment by fharper** [comment]
My app don't crash per say, but it freeze when scanning the QR code. Deleting the app and reinstalling didn't fix it.

**Comment by Miguel-Signal** [comment]
@fharper Thanks for letting us know. Could you recreate the issue then write into [support@signal.org](mailto:support@signal.org) with a debug log link?

To share a debug log from Signal iOS, tap your profile, Settings, Help, Submit Debug Log, Copy Link.

**Comment by fharper** [comment]
> [@fharper](https://github.com/fharper) Thanks for letting us know. Could you recreate the issue then write into [support@signal.org](mailto:support@signal.org) with a debug log link?
> 
> To share a debug log from Signal iOS, tap your profile, Settings, Help, Submit Debug Log, Copy Link.

Done
